vowpalwabbit/prob_dist_cont.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "vw.h"
+#include "v_array.h"
+#include "prob_dist_cont.h"
+#include "io_buf.h"
+
+using namespace std;
+// TODO: below check
+namespace VW { namespace actions_pdf
+{
+  void delete_prob_dist(void* v)
+  {
+    v_array<pdf_segment>* cs = (v_array<pdf_segment>*)v;
+    cs->delete_v();
+  }
+
+  std::string to_string(const action_pdf_value& a_p, bool newline)
```

Comment:
Not sure I understand why this function needs conditionally add a newline. Isn't it simpler to do that at the callsite?

Reply:
I was on the fence on this.  Adding it in the calling site (writing predictions to file)  requires either a buffer copy or a bit of ugly code and accounting for platform differences in new line.  The parameter along with the default of newline=false seems like the best alternative.